core-tests/src/test/java/org/mule/runtime/core/processor/MapProcessorTestCase.java

```diff
@@ -0,0 +1,138 @@
+import static org.mule.tck.junit4.AbstractMuleContextTestCase.RECEIVE_TIMEOUT;
+import static reactor.core.publisher.Mono.just;
+import org.mule.runtime.core.api.Event;
+import org.mule.runtime.core.api.EventContext;
+import org.mule.runtime.core.api.processor.Processor;
+import org.mule.runtime.core.util.concurrent.Latch;
+import org.mule.tck.junit4.AbstractMuleTestCase;
+import org.mule.tck.size.SmallTest;
+
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+@SmallTest
+public class MapProcessorTestCase extends AbstractMuleTestCase {
+
+  @Mock
+  private EventContext eventContext;
+
+  @Mock
+  private Event event;
+
+  @Test
+  public void mapBlocking() throws Exception {
+    Processor mapProcessor = event -> Event.builder(eventContext).message(of(TEST_PAYLOAD)).build();
+    assertThat(mapProcessor.process(event).getMessage().getPayload().getValue(), equalTo(TEST_PAYLOAD));
+  }
+
+  @Test
+  public void mapStreamBlockingGet() {
+
+    Processor mapProcessor = event -> Event.builder(eventContext).message(of(TEST_PAYLOAD)).build();
+
+    assertThat(just(event).transform(mapProcessor).block().getMessage().getPayload().getValue(),
+               equalTo(TEST_PAYLOAD));
+  }
+
+  @Test
+  public void mapStreamSubscribe() throws Exception {
+    Processor mapProcessor = event -> Event.builder(eventContext).message(of(TEST_PAYLOAD)).build();
+    Latch latch = new Latch();
+    just(event).transform(mapProcessor).subscribe(event -> {
+      latch.countDown();
+      assertThat(event.getMessage().getPayload().getValue(), equalTo(TEST_PAYLOAD));
+    });
+    latch.await(RECEIVE_TIMEOUT, TimeUnit.MILLISECONDS);
+  }
+
+  @Test
+  public void mapBlockingNullResult() throws Exception {
+    Processor mapProcessor = event -> Event.builder(eventContext).message(of(null)).build();
+    assertThat(mapProcessor.process(event).getMessage().getPayload().getValue(), equalTo(null));
+  }
+
+  @Test
+  public void mapStreamBlockingGetNullResult() {
+
+    Processor mapProcessor = event -> Event.builder(eventContext).message(of(null)).build();
+
+    assertThat(just(event).transform(mapProcessor).block().getMessage().getPayload().getValue(),
+               equalTo(null));
+  }
+
+  @Test
+  public void mapStreamSubscribeNullResult() throws Exception {
+    Processor mapProcessor = event -> Event.builder(eventContext).message(of(null)).build();
+    Latch latch = new Latch();
+    just(event).transform(mapProcessor).subscribe(event -> {
+      latch.countDown();
+      assertThat(event.getMessage().getPayload().getValue(), equalTo(null));
+    });
+    latch.await(RECEIVE_TIMEOUT, TimeUnit.MILLISECONDS);
+  }
+
+  @Test
+  public void mapBlockingExceptionThrown() throws Exception {
+    Processor mapProcessor = event1 -> {
+      throw new IllegalArgumentException();
```

Comment:
Can yo uchange this to use an exception type specific for this test case? Just paraniod about the reactor code throwing this at some point.